### DISCUSSION TITLE
Rank score penalty points

### DISCUSSION THREAD

**Discussion by 1ctw** [Clinical-Genomics/scout#5295] [Q&A]
I have a question regarding rank score penalty points which I assume is due to lack of "good enough" compound.

I found a missense variant and a SV (small deletion) in the gene FBP1 in our case 24333:

https://scout.scilifelab.se/cust003/24333/e62ee82dc2c3c093dabbc48f3b445232

https://scout.scilifelab.se/cust003/24333/sv/variants/8d69be7ec28a00d654cc6218e026dcfd

The missense variant has 16 points in the "Rank score results" but gets only "Rank score" 10 and the SV has 12 points in the "Rank score results" but gets a "Rank score" of 6. 

The missense variant has been found in homozygous state earlier and then had a "Rank score" of 16, see https://scout.scilifelab.se/cust003/22425/3b1017a6d3a778c169c093e6617a1fad.

It seems that both the missense variant and deletion get 6 minus points. Are they not connected as possible compounds resulting in all these points being deducted for both variants? There are also other possible SNV compounds in this case but they are intronic with pretty low "Rank score".  What is the limit in terms of rank score for the variants to receive these 6 minus points? Can you please explain how this penalty system is working. I am pretty sure those two variants are causative in case 24333 and I'm not happy that they get such a low "Rank score".

**Comment by dnil** [comment]
Hi, we have explained this a few times, but let's go again; I also didn't find it quickly among the old issues. 

The default low scoring compound threshold is 10 (so variants from 9 and down are low, giving the default -6 penalty). Again, I very much think this is not an ideal situation. As mentioned on the meeting today, we have decided to more actively maintain genmod for at least another year. This is high up on the list of issues that need solving - feel free to chime in on https://github.com/Clinical-Genomics/genmod/issues/107.

There is currently no compound connection between variants of fundamentally different types, eg STR vs SV vs SNV - the best we do is SV overlap, but that is in scout and after ranking.

I quite agree with you; the del is clear, and while the missense one is a little difficult on its own, it would deserve at least its score 16 in this case. This is yet another reason why the "missing compound" penalty isn't a great idea.  We should be adjusting the other scores to add some nuance with these variants if they are too abundant.

I think we should also attribute the SV more score in general, even before the -6 malus. In this case, a small frequency malus is in order since it is recessive, and present eg in gnomAD-SV. It is high quality, found by all callers.
It seems unfortunately not to be in ClinVar yet, or at least not at the date of the ClinVar dump?
The main issue is presumably the blandness of the VEP functional annotation/most sever consequence script. This is a full first exon ablation, and should be given a much harsher verdict than "splice polypyrimidine tract variant". @jemten @ramprasadn what happens to this SV with the new VEP version and consequence script in nf-core/raredisease? Perhaps annotSV or one of its cousins will help?

This particular one will of course be found in the future now that you flag it causative, and later once you export it to clinvar, but we should try to cater to similar variants in other genes.

**Comment by jemten** [comment]
Interesting and unfortunate that the DEL penalised. I will re-run this case with the new pipeline and see what happens. In the new pipe we are running a more recent version of VEP and use a different script to pick out the most severe consequence. I'll get back to you

**Comment by 1ctw** [comment]
@dnil I´m sorry for asking again, I remember that we have discussed similar issues before, but infortunately I didn´t rembember the details of the penalty system. Thanks for explaining:) However, good to discuss again.

It seems not to be in ClinVar yet, even if I found several publications describing this deletion in HGMD. It will be in ClinVar and set as causative as soon as we have segregated the variants and confirmed the defect biochemically, but it seems to fit perfectly with the clinical picture.

Thanks @jemten for a re-run of this case with the new pipeline to see what happens.

**Comment by dnil** [comment]
No worries, thank you for the patience: this is a great example for several reasons. I encourage many questions here: the compound penalty is a dubious solution to a problem that may or may not exist any more. And the SV annotation/scoring has quite some room for improvement - some of which has happened already in development of the new pipeline.

**Comment by dnil** [comment]
Ping @jemten - do you remember what this gave? Also, I still very much think we should spend the effort to remove that -6. It keeps recurring for most of the problematic variants in all "what would be the optimal score threshold for my panel" type of analysis.

**Comment by jemten** [comment]
Sorry I forgot to report on this. 
For the deletion: VEP annotated the most sever consequence as "feature_truncation", which is quite high on the severity list of VEP's [consequence list](https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html) and more in line with transcript ablation. However in the rankmodel we have listed the feature truncation consequence as not that interesting. This is not something new with this version but rather something we've had since a very long time (the beginning?) and should be re-evaluated. Right now this results in the variant having an even lower rankscore than before. 

@dnil, I'm inclined to skip our custom ranking of feature severity and rely on VEP's list instead. What do you think?
